### PR TITLE
[2380] Time Needed to Rearrange a Binary String

### PR DESCRIPTION
**Pull request by chhavibansal**
https://leetcode.com/problems/time-needed-to-rearrange-a-binary-string/

### CODE REVIEW

File: LeetcodeProblemsTests/Algorithms/Time_Needed_Rearrange_Binary_String_Test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const assert = require('assert');
+const secondsToRemoveOccurrences = require('../../LeetcodeProblems/Algorithms/Time_Needed_Rearrange_Binary_String').secondsToRemoveOccurrences;
+
+function test() {
+  assert.equal(secondsToRemoveOccurrences("0110101"));
+  assert.equal(secondsToRemoveOccurrences("11100"))
```
Comment (ignacio-chiazzo) [inline LeetcodeProblemsTests/Algorithms/Time_Needed_Rearrange_Binary_String_Test.js:6]: could you add the assert equals to a number?

**Review by ignacio-chiazzo — APPROVED** [review]
LGTM! 

I'm curious if there is a solution with less order than O(n**2).

File: LeetcodeProblemsTests/Algorithms/Time_Needed_Rearrange_Binary_String_Test.js (hunk 1 above)
Comment (chhavibansal) [reply LeetcodeProblemsTests/Algorithms/Time_Needed_Rearrange_Binary_String_Test.js]: i have added assertion, also i saw there was assertion missing for my last PR added that change as well. 

also this is  O ( N ) solution

Comment (ignacio-chiazzo) [reply LeetcodeProblemsTests/Algorithms/Time_Needed_Rearrange_Binary_String_Test.js]: The `replaceAll` would have to iterate over the whole string, and the while(true) iterates over it too (it is not fixed 🤔 ).

**Comment by ignacio-chiazzo** [comment]
Thank you!

**Pull request merged**